### PR TITLE
nemotron config exp

### PR DESCRIPTION
**Pull request by ved1beta**
```
100%|_____________________________________________________________________________________________________________________________________| 324/324 [00:01<00:00, 200.47it/s]
[2026-03-17 06:39:19,076] [INFO] [axolotl.loaders.model] converting PEFT model w/ prepare_model_for_kbit_training
[2026-03-17 06:39:19,080] [INFO] [axolotl.loaders.model] Converting modules to torch.bfloat16
[2026-03-17 06:39:19,086] [INFO] [axolotl.loaders.adapter] found linear modules: ['down_proj', 'k_proj', 'o_proj', 'q_proj', 'up_proj', 'v_proj']
trainable params: 46,137,344 || all params: 4,236,652,544 || trainable%: 1.0890
[2026-03-17 06:39:22,249] [INFO] [axolotl.train] Pre-saving adapter config to ./outputs/nemotron-mini-4b-qlora...
[2026-03-17 06:39:22,256] [INFO] [axolotl.train] Pre-saving tokenizer to ./outputs/nemotron-mini-4b-qlora...
[2026-03-17 06:39:22,749] [INFO] [axolotl.train] Pre-saving model config to ./outputs/nemotron-mini-4b-qlora...
[2026-03-17 06:39:22,768] [INFO] [axolotl.train] Starting trainer...
The tokenizer has new PAD/BOS/EOS tokens that differ from the model config and generation config. The model config and generation config were aligned accordingly, being updated with the tokenizer's values. Updated tokens: {'pad_token_id': 256000}.
[2026-03-17 06:39:27,927] [INFO] [axolotl.utils.samplers.multipack] gather_len_batches: [34]
{'loss': '1.39', 'grad_norm': '0.09095', 'learning_rate': '0.0002', 'ppl': '4.014', 'memory/max_active (GiB)': '31', 'memory/max_allocated (GiB)': '31', 'memory/device_reserved (GiB)': '36.39', 'tokens/train_per_sec_per_gpu': '900.4', 'tokens/total': 32768, 'tokens/trainable': 27228, 'epoch': '0.1176'}
{'loss': '1.443', 'grad_norm': '0.1009', 'learning_rate': '0.0001924', 'ppl': '4.235', 'memory/max_active (GiB)': '31.08', 'memory/max_allocated (GiB)': '31.08', 'memory/device_reserved (GiB)': '36.48', 'tokens/train_per_sec_per_gpu': '1034', 'tokens/total': 65536, 'tokens/trainable': 53948, 'epoch': '0.2353'}
{'loss': '1.238', 'grad_norm': '0.0908', 'learning_rate': '0.0001707', 'ppl': '3.447', 'memory/max_active (GiB)': '31.08', 'memory/max_allocated (GiB)': '31.08', 'memory/device_reserved (GiB)': '36.48', 'tokens/train_per_sec_per_gpu': '1001', 'tokens/total': 98304, 'tokens/trainable': 81512, 'epoch': '0.3529'}
{'loss': '1.225', 'grad_norm': '0.07909', 'learning_rate': '0.0001383', 'ppl': '3.403', 'memory/max_active (GiB)': '31.08', 'memory/max_allocated (GiB)': '31.08', 'memory/device_reserved (GiB)': '36.48', 'tokens/train_per_sec_per_gpu': '1030', 'tokens/total': 131072, 'tokens/trainable': 108594, 'epoch': '0.4706'}
{'loss': '1.139', 'grad_norm': '0.03827', 'learning_rate': '0.0001', 'ppl': '3.123', 'memory/max_active (GiB)': '31.08', 'memory/max_allocated (GiB)': '31.08', 'memory/device_reserved (GiB)': '36.48', 'tokens/train_per_sec_per_gpu': '983.7', 'tokens/total': 163840, 'tokens/trainable': 135522, 'epoch': '0.5882'}
{'loss': '1.127', 'grad_norm': '0.03183', 'learning_rate': '6.173e-05', 'ppl': '3.087', 'memory/max_active (GiB)': '31.08', 'memory/max_allocated (GiB)': '31.08', 'memory/device_reserved (GiB)': '36.48', 'tokens/train_per_sec_per_gpu': '950.4', 'tokens/total': 196608, 'tokens/trainable': 162092, 'epoch': '0.7059'}
{'loss': '1.113', 'grad_norm': '0.03033', 'learning_rate': '2.929e-05', 'ppl': '3.044', 'memory/max_active (GiB)': '31.08', 'memory/max_allocated (GiB)': '31.08', 'memory/device_reserved (GiB)': '36.48', 'tokens/train_per_sec_per_gpu': '1045', 'tokens/total': 229376, 'tokens/trainable': 189030, 'epoch': '0.8235'}
{'loss': '1.123', 'grad_norm': '0.03211', 'learning_rate': '7.612e-06', 'ppl': '3.074', 'memory/max_active (GiB)': '31.08', 'memory/max_allocated (GiB)': '31.08', 'memory/device_reserved (GiB)': '36.48', 'tokens/train_per_sec_per_gpu': '1005', 'tokens/total': 262144, 'tokens/trainable': 215803, 'epoch': '0.9412'}
100%|___________________________________________________________________________________________________________________________________________________________| 8/8 [00:56<00:00,  6.83s/it]
[2026-03-17 06:40:24,525] [INFO] [axolotl.core.trainers.base] Saving model checkpoint to ./outputs/nemotron-mini-4b-qlora/checkpoint-8
[2026-03-17 06:40:24,771] [WARNING] [py.warnings] /root/miniconda3/envs/py3.11/lib/python3.11/site-packages/peft/utils/save_and_load.py:309: UserWarning: Setting `save_embedding_layers` to `True` as the embedding layer has been resized during finetuning.
  warnings.warn(

{'train_runtime': '69.3', 'train_samples_per_second': '0.923', 'train_steps_per_second': '0.115', 'train_loss': '1.225', 'memory/max_active (GiB)': '4.47', 'memory/max_allocated (GiB)': '4.47', 'memory/device_reserved (GiB)': '36.48', 'epoch': '0.9412', 'tokens/train_per_sec_per_gpu': '0'}
100%|___________________________________________________________________________________________________________________________________________________________| 8/8 [01:09<00:00,  8.66s/it]
[2026-03-17 06:40:37,361] [INFO] [axolotl.train] Training completed! Saving trained model to ./outputs/nemotron-mini-4b-qlora.
[2026-03-17 06:40:37,612] [WARNING] [py.warnings] /root/miniconda3/envs/py3.11/lib/python3.11/site-packages/peft/utils/save_and_load.py:309: UserWarning: Setting `save_embedding_layers` to `True` as the embedding layer has been resized during finetuning.
  warnings.warn(
```

nvidia/NVIDIA-Nemotron-3-Super-120B-A12B NEXT

## Release Notes

* **New Features**
  * Added support for Nemotron-Mini-4B-Instruct model training with QLoRA adapters for efficient fine-tuning.
  * Included comprehensive training configuration example with 4-bit quantization, gradient accumulation, cosine learning rate scheduling, warmup, flash attention optimization, and experiment tracking integration.

### CODE REVIEW

File: examples/nemotron/nemotron-mini-4b-qlora.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+resume_from_checkpoint:
+logging_steps: 1
+flash_attention: true
+
+warmup_ratio: 0.1
+evals_per_epoch: 1
+saves_per_epoch: 1
+
+special_tokens:
+  pad_token: <|endoftext|>
```
Comment (NanoCode012) [inline examples/nemotron/nemotron-mini-4b-qlora.yaml]:
```suggestion
```

This model uses `<pad>` https://huggingface.co/nvidia/Nemotron-Mini-4B-Instruct/blob/main/tokenizer_config.json#L4

No need to override this

**Review by NanoCode012 — APPROVED** [review]

**Review by winglian — APPROVED** [review]
Need to add the architecture to `MOE_ARCH_BLOCK` as well for zero3 support. Good to merge once that's updated.

**Comment by NanoCode012** [comment]
> Need to add the architecture to `MOE_ARCH_BLOCK` as well for zero3 support. Good to merge once that's updated.

This is not the MoE one. This is a year old. We accidentally got it confused and tested this, so added it anyways.
